internal/sgbouncewizard/sgbouncewizard.go

```diff
@@ -33,55 +41,68 @@ func NewServer(c config.Configuration) (*Server, error) {
 func (srv *Server) RuleContext(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		var rule *models.BounceRule
-		var err error
-		if bounce_id := chi.URLParam(r, "bounce_id"); bounce_id != "" {
-			rule, err = srv.DBClient.GetRuleDB(bounce_id)
+
+		bounce_id := chi.URLParam(r, "bounce_id"); 
+		bouncd_idInt, err := strconv.Atoi(bounce_id)
+		
+		if err != nil {
+			log.Println(err)
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if bounce_id != "" {
+			rule, err = srv.DBClient.GetSingleRule(bouncd_idInt)
 			if err != nil {
-				http.Error(w, err.Error(), http.StatusInternalServerError)
+				log.Println(err)
+				http.Error(w, err.Error(), http.StatusNotFound)
+				return
 			}
 		}
 		ctx := context.WithValue(r.Context(), "rule", rule)
 		next.ServeHTTP(w, r.WithContext(ctx))
 	})
 }
 
-func (srv *Server) generateHash(pwd []byte) string {
+func (srv *Server) generateHash(pwd []byte) (*string, error) {
```

Comment:
Why return pointer to string?

Reply:
if it fails I want to return nill and it was giving men and error that nill can't be assigned to string

Reply:
Correct nil can't be assigned to string, but in case of an error you can return the zero value for string which is empty string. In this case just `return "", errors.Wrap(err,"failed to gen hash")`